vaadin-checkbox-flow-parent/vaadin-checkbox-testbench/src/main/java/com/vaadin/flow/component/checkbox/testbench/CheckboxGroupElement.java

```diff
@@ -0,0 +1,147 @@
+     * @param label
+     *            the label of the checkbox to select
+     */
+    public void selectByText(String label) {
+        Optional<CheckboxElement> checkbox = getCheckboxByLabel(label);
+        if (!checkbox.isPresent()) {
+            throw new NoSuchElementException(
+                    "No checkbox with label '" + label + "' found");
+        }
+
+        checkbox.get().setChecked(true);
+    }
+
+    /**
+     * Attempts to deselect a checkbox that is currently selected, by matching
+     * the label. Does nothing if the checkbox is not selected.
+     *
+     * @throws NoSuchElementException
+     *             if no matching label was found
+     * @param label
+     *            the label of the checkbox to select
+     */
+    public void deselectByText(String label) {
+        Optional<CheckboxElement> checkbox = getCheckboxByLabel(label);
+        if (!checkbox.isPresent()) {
+            throw new NoSuchElementException(
+                    "No checkbox with label '" + label + "' found");
+        }
+
+        checkbox.get().setChecked(false);
+    }
+
+    /**
+     * Gets the labels of the currently selected checkboxes.
+     *
+     * @return the labels of the currently selected checkboxes
+     */
+    public List<String> getSelectedTexts() {
+        Stream<CheckboxElement> button = getSelectedCheckboxes();
+        return button.map(CheckboxElement::getLabel)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets the selected checkboxes.
+     *
+     * @return a stream with the selected checkboxes elements
+     */
+    private Stream<CheckboxElement> getSelectedCheckboxes() {
+        return getCheckboxByChecked(true);
+    }
+
+    /**
+     * Gets the checkbox which is part of this group and has the given label.
+     *
+     * @param label
+     *            the label to search for
+     * @return an optional with the checkbox element or an empty optional if no
+     *         match was found
+     */
+    private Optional<CheckboxElement> getCheckboxByLabel(String label) {
+        if (label == null) {
+            return Optional.empty();
+        }
+        return getCheckboxes().stream()
+                .filter(checkbox -> label.equals(checkbox.getLabel()))
+                .findFirst();
+    }
+
+    /**
+     * Gets the stream of checkboxes which are part of this group and has the
+     * checked state.
+     *
+     * @param value
+     *            the value to search for
+     * @return a stream with the checkbox elements
+     */
+    private Stream<CheckboxElement> getCheckboxByChecked(boolean value) {
+        return getCheckboxes().stream()
+                .filter(checkbox -> checkbox.isChecked() == value);
+    }
+
+    /**
+     * Gets the slotted error message component for the element.
+     *
+     * @return the slotted component or {@code null} if there is no component
+     */
+    public TestBenchElement getErrorMessageComponent() {
```

Comment:
I'm afraid these methods won't work in v14 because they have been designed with the v24 HTML structure in mind.  Considering that, I tend to be in favor of not introducing `CheckboxGroupElement` at all. In tests, it's still possible to get the element reference with `$("vaadin-checkbox-group")`, if we use a general `TestBenchElement` instead of `CheckboxGroupElement` in the class type.